rllib/utils/actors.py

```diff
@@ -25,36 +25,18 @@ def add(self, worker, all_obj_ids):
         self._tasks[obj_id] = worker
         self._objects[obj_id] = all_obj_ids
 
-    def completed(self, blocking_wait=False):
+    def completed(self, blocking_wait=False, max_yield=999):
         pending = list(self._tasks)
+        num_yielded = 0
         if pending:
             ready, _ = ray.wait(pending, num_returns=len(pending), timeout=0)
             if not ready and blocking_wait:
                 ready, _ = ray.wait(pending, num_returns=1, timeout=10.0)
             for obj_id in ready:
+                if (num_yielded >= max_yield):
+                    break
```

Comment:
Ah, this is wrong since you are discarding data...

Reply:
Am I? Object IDs only get popped from self._tasks below, so they will still get `wait`'d on subsequent calls. Might me missing something.